global.json

```diff
@@ -1,5 +1,5 @@
 {
-  "sdk": {
-    "version": "2.1.401"
+  "tools": {
+    "dotnet": "2.1.401"
   }
 }
```

Comment:
Can you explain this change? Won't the SDK just ignore global.json now?

Reply:
That's correct. The goal was to avoid forcing SDK version via global.json. 
This is now only for Arcade dotnet restore/finding logic. See https://github.com/dotnet/arcade/issues/320

Reply:
So what will the experience be when someone tries to build Roslyn.sln but doesn't have the proper .NET SDK installed on the machine?

Reply:
@nguerrera

Reply:
We could hook up `InitializeDotNetToolPath` target to `Restore` target, so that when restore is attempted and dotnet SDK of the right version is not found we report an error.

Reply:
There isn't a good built-in today experience for specifying a *minimum* version of the sdk. sdk/version in global.json pins the SDK (modulo patch) instead. As I explained in linked issue, this hurts dogfooding tremendously. Our devs should be using the SDK that came with their dogfooding channel VS wherever possible. That is currently 2.1.5xx. Pinning the sdk to 2.1.4xx prevents that. It also unnecessarily requires me to install 2.1.4xx when 2.1.5xx would work.

If you want to enforce minimum version, you could do that in a target in various ways. If you want to actually pin despite the downsides and my recommendationa against it, then you can keep sdk/version in the global.json. My highest order point in the linked issue is that we shouldn't be reusing the pinned sdk field in global.json to represent the sdk that is acquired for command-line / official builds, and that is solved. sdk/version is for pinning, tools/dotnet is for acquisition. There is nothing technically stopping you from using both, but I have found pinning to do more harm than good in our projects.

Reply:
The target `InitializeDotNetToolPath` is currently looking for exact version match. I can update it to find exact or higher version. Then we can hook it up to `Restore` and report an error.

Reply:
I agree a minimum requirement is better here than exact. At the same time I end up fielding all of the emails for:

> jared I tried to build Roslyn and got so many  errors that my machine literally caught fire and evaporated my coffee. I can take the machine being destroyed but the loss of coffee is something I'll never get over.

I don't understand why it's so impossible for us to get a simple

Reply:
Updated PR to check .NET SDK version before build